### PR TITLE
Fix unit test in BTAnalyticsService_Tests

### PR DESCRIPTION
**Pull request by warmkesselj**
### Summary of changes

DTBTSDK-3514

This test has been failing consistently: testSendAnalyticsEvent_whenNumberOfQueuedEventsMeetsThreshold_sendsAnalyticsEvent

### Checklist

- [ ] Added a changelog entry

### CODE REVIEW

**Comment by scannillo** [comment]
Edited the PR description to remove the JIRA link!

**Review by sshropshire — APPROVED** [review]
🌊

File: UnitTests/BraintreeCoreTests/Analytics/BTAnalyticsService_Tests.swift [hunk 1]
```diff
@@ -35,19 +35,23 @@ final class BTAnalyticsService_Tests: XCTestCase {
...
         
-        let timestamp = parseTimestamp(mockAnalyticsHTTP.lastRequestParameters)!
-        let eventName = parseEventName(mockAnalyticsHTTP.lastRequestParameters)
-        XCTAssertEqual(eventName, "an.analytics.event")
-        XCTAssertGreaterThanOrEqual(timestamp, currentTime)
-        XCTAssertLessThanOrEqual(timestamp, oneSecondLater)
-        validateMetadataParameters(mockAnalyticsHTTP.lastRequestParameters)
+        analyticsService.sendAnalyticsEvent("any.analytics.event") { error in
+            expectation.fulfill()
+        }
```
Comment (richherrera) [inline UnitTests/BraintreeCoreTests/Analytics/BTAnalyticsService_Tests.swift:42]: do we need  parameter`error`? maybe we can change it to `_` or validate if the error is different to nil
```suggestion
        analyticsService.sendAnalyticsEvent("any.analytics.event") { error in
            if let error {
               // assert
            } else {
               expectation.fulfill()
            }
        }
```

Comment (richherrera) [reply UnitTests/BraintreeCoreTests/Analytics/BTAnalyticsService_Tests.swift:42]: same question for all the events below with error parameter

**Review by richherrera — APPROVED** [review]
LGTM!!

File: UnitTests/BraintreeCoreTests/Analytics/BTAnalyticsService_Tests.swift (hunk 1 above)
Comment (warmkesselj) [reply UnitTests/BraintreeCoreTests/Analytics/BTAnalyticsService_Tests.swift:42]: Good catch. I just merged and I can make a change in a follow up PR

**Pull request merged**